runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -126,8 +127,18 @@ public CheckpointWriter(CorfuRuntime rt, UUID streamId, String author, T map) {
      */
     public Token appendCheckpoint() {
         long start = System.currentTimeMillis();
+
+        // Queries the sequencer for the global log tail. We then read the global log tail to
+        // persist the entry on the logunit in turn preventing from a new sequencer from
+        // regressing tokens.
+        Token markerToken = rt.getSequencerView().query().getToken();
+        if (Address.nonAddress(markerToken.getSequence())) {
+            return markerToken;
```

Comment:
Is it correct to return this token? appendCheckpoint is expecting the Token for the checkpoint and it is not true that this is it, we did not even checkpoint, should we throw an error? or is this our contract?

Reply:
I would rather increment the stream tail to prevent regression because that's correct "by-construction", but this method assumes that no writer will be able to write between the stream tail and the marker, which is still error prone.  

Anyways, we already had this discussion just wanted to put it on the record.